### PR TITLE
Mat 1461

### PR DESCRIPTION
**Pull request by imyourm8**

### CODE REVIEW

**Comment by jdkanani** [comment]
@imyourm8 Please resolve conflict. Looks good otherwise.

File: contracts/staking/validatorShare/ValidatorShare.sol [hunk 1]
```diff
@@ -212,13 +212,20 @@ contract ValidatorShare is IValidatorShare {
...
+            return 0;
+        }
+        
+        uint256 liquidRewards;
+        uint256 totalTokens = exchangeRate().mul(share).div(100);
+        if (totalTokens >= amountStaked[user]) {
+            liquidRewards = totalTokens.sub(amountStaked[user]);
+        }
+
+        return liquidRewards;
```
Comment (atvanguard) [inline contracts/staking/validatorShare/ValidatorShare.sol:228]: Less code -> easier on the eyes and saved gas.
```suggestion
        uint256 totalTokens = exchangeRate().mul(share).div(100);
        if (totalTokens >= amountStaked[user]) {
            return totalTokens.sub(amountStaked[user]);
        }
        return 0;
```

File: test/units/staking/ValidatorShare.test.js [hunk 2]
```diff
@@ -695,4 +695,24 @@ contract('ValidatorShare', async function() {
       })
     })
   })
+
+  describe.only('getLiquidRewards', function() {
```
Comment (atvanguard) [inline test/units/staking/ValidatorShare.test.js:699]: Remember to remove `.only` before merge and check for CI

File: contracts/staking/validatorShare/ValidatorShare.sol (hunk 1 above)
Comment (imyourm8) [reply contracts/staking/validatorShare/ValidatorShare.sol:228]: I would argue that your version uses less gas. With early exit i don't read any of the storage variables. 32 bytes of memory expansion cost 1 gas, so in general my implementation is more efficient, i only need to read amountStaked[user] once.

Comment (atvanguard) [reply contracts/staking/validatorShare/ValidatorShare.sol]: Only case where your version requires less gas is when shares == 0. I personally like to optimize for the usual case, not an outlier.

**Pull request merged**